packages/llamaindex/src/tools/AzureDynamicSessionTool.ts

```diff
@@ -0,0 +1,355 @@
+import {
+  DefaultAzureCredential,
+  getBearerTokenProvider,
+} from "@azure/identity";
+import { getEnv } from "@llamaindex/env";
+import crypto from "node:crypto";
```

Comment:
node dependencies are capsulated in `@llamaindex/env` (so `llamaindex` is working with edge, cloudflare, deno etc.)

we either need to encapsulate the deps or move the tool to another location. I suggest https://github.com/run-llama/create-llama/tree/main/templates/components/engines/typescript/agent/tools then I can also add this tool to create-llama and we can make a demo

Reply:
Happy to move the `crypto` dep to llamaindex/env. We only use it to get a random uuidv4.

Reply:
great. keep in mind that the other `node:*` deps also need to be moved to support different platforms

Reply:
do you guys have a process you use to test llamaindex/env across all supported environments?

Reply:
we have a couple of e2e tests in https://github.com/run-llama/LlamaIndexTS/tree/main/packages/llamaindex/e2e/examples

Reply:
they are also running with CI

Reply:
@himself65 suggested that I rename `AzureDynamicSessionTool.ts` to `AzureDynamicSessionTool.node.ts`. For this initial release, I think it's fine to only target node. We'd definetly target other runtimes per users requests.